pkg/utils/kubernetes/ownerrefs.go

```diff
@@ -38,6 +42,40 @@ func SetOwnerForObject(obj, owner client.Object) {
 	}
 }
 
+// SetOwnerForObjectThroughLabels sets the owner of the provided object through a label.
+func SetOwnerForObjectThroughLabels(obj, owner client.Object) error {
+	labels := obj.GetLabels()
+	managedByLabelSet, err := GetManagedByLabelSet(owner)
+	if err != nil {
+		return err
+	}
+	for k, v := range managedByLabelSet {
+		labels[k] = v
+	}
+	obj.SetLabels(labels)
+	return nil
+}
+
+// GetManagedByLabelSet returns a map of labels with the managing object's metadata.
+func GetManagedByLabelSet(obj client.Object) (map[string]string, error) {
```

Comment:
- Food for thought: we could limit the types accepted by this func via generic type constraint and this way:
  
  - not loose type information
  - not need the `error` return type

- How about adding a simple UT for this to guard the functionality from changing unexpectedly?

Reply:
yeah, it makes total sense. I've added the generics and a simple UT 👍